### PR TITLE
Ports: Add default zshrc

### PR DESCRIPTION
**Pull request by fdellwing**
Previosly we had a very messed up PS1 and no common aliases.

We now provide a default `zshrc` file for the system that uses sane values to make the port more usable.

Before:

grafik

After:

grafik

### CODE REVIEW

**Comment by nico** [comment]
This zhsrc does a lot more than setting a default prompt (Auto complete stuff, `alias pu=pushd`). That seems kind of out of scope for a system zhsrc file -- shouldn't we honor all defaults except for the ones that cause bugs for some reason (and have a comment about the big they work around for those)?

**Comment by fdellwing** [comment]
That is a possibility, yes.

I opted to provide a few more nice to have values that are part of most examples. If that is not what we want for it, I will reduce it to the absolute minimum to fix the prompt (that will be a single line).

File: Ports/.gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@
 !*/patches/*
 !*/.gitignore
 !*/README.md
+!zsh/zshrc
```
Comment (timschumi) [inline Ports/.gitignore:7]: This should probably just go into a new, zsh-specific `.gitignore` file.

File: Ports/zsh/zshrc [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+alias cp='nocorrect cp'       # no spelling correction on cp
+alias mkdir='nocorrect mkdir' # no spelling correction on mkdir
+alias j=jobs
+alias pu=pushd
+alias po=popd
+alias d='dirs -v'
+alias h=history
+alias grep=egrep
+alias ll='ls -l'
+alias la='ls -a'
```
Comment (timschumi) [inline Ports/zsh/zshrc:30]: In my opinion, if we want to add some default aliases, we should ideally mirror at most the ones that already exist in our own Shell's rc file.

In fact, that should probably apply to all settings that we could set here that have equivalent functionality in the default system.

**Review by timschumi — CHANGES_REQUESTED** [review]

File: Ports/zsh/zshrc (hunk 2 above)
Comment (fdellwing) [reply Ports/zsh/zshrc:30]: Ok, I left in a single alias, basic autocompletion and history management as well as the fix for the prompt.

File: Ports/zsh/zshrc [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+###########
+# History #
+###########
+
+# number of lines kept in history
+export HISTSIZE=1000
+# number of lines saved in the history after logout
+export SAVEHIST=1000
+# location of history
+export HISTFILE=~/.zsh_history
```
Comment (emanuele6) [inline Ports/zsh/zshrc:34]: These should definitely **not** be exported: there is no reason to export them, and exporting them would just possibly break the history of `bash` or other shells (and likely also overwrite, corrupt, or wipe `~/.zsh_history`) if you run them from `zsh` and if they don't overwrite those variables in their bashrc file or equivalent.

**Review by emanuele6 — CHANGES_REQUESTED** [review]

File: Ports/zsh/zshrc (hunk 3 above)
Comment (fdellwing) [reply Ports/zsh/zshrc:34]: Good catch!

File: Ports/zsh/zshrc [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+# Case-insensitive (all),partial-word and then substring completion
+zstyle ':completion:*' matcher-list 'm:{a-zA-Z}={A-Za-z}' 'r:|[._-]=* r:|=*' 'l:|=* r:|=*'
+
+###########
+# Aliases #
+###########
+
+# Set up aliases
+alias ll='ls -l'
```
[the remark this answers is not in the document]
Comment (fdellwing) [reply Ports/zsh/zshrc:20]: You are aware, that our default shellrc also sets up things like autocomplete?

[the remark this answers is not in the document]
Comment (nico) [reply Ports/zsh/zshrc:20]: Unrelated to the messy screenshot up top :)

File: Ports/zsh/package.sh [hunk 5]
```diff
@@ -13,3 +13,7 @@ pre_configure() {
 post_configure() {
     run_replace_in_file "s/define HAVE_PRCTL 1/undef HAVE_PRCTL/" config.h
 }
+
+post_install() {
+    run cp "${PORT_META_DIR}/zshrc" "${SERENITY_INSTALL_ROOT}/etc/"
```
[the remark this answers is not in the document]
Comment (fdellwing) [reply Ports/zsh/package.sh:18]: No need at all to do `run` as we dont care about the path, so we just skip the wrapper completely.

**Pull request merged**